File: AlgorithmJavaVersion/src/DataStructure/heapStackQueue/queue/LinkedDeQueuelj.java
package DataStructure.heapStackQueue.queue;

import DataStructure.heapStackQueue.DeQueuelj;

/**
 * @author 蔚蔚樱
 * @version 1.0
 * @date 2022/4/6 12:09
 * @author—Email micromicrohard@outlook.com
 * @description 链表实现 双端队列
 * @blogURL
 */

public class LinkedDeQueuelj implements DeQueuelj {

    
    public boolean offerFront(int value) {
        return false;
    }

    
    public int pollFront() {
        return poll();
    }

    
    public int peekFront() {
        return peek();
    }

    
    public boolean offerTail(int value) {
        return offer(value);
    }

    
    public int pollTail() {
        return 0;
    }

    
    public int peekTail() {
        return 0;
    }

    
    public boolean offer(int value) {
        return false;
    }

    
    public int poll() {
        return 0;
    }

    
    public int peek() {
        return 0;
    }

    
    public boolean empty() {
        return false;
    }

    
    public int getRealSize() {
        return 0;
    }

    
    public int getMaxSize() {
        return 0;
    }

    
    public void resize() {
    }

    
    public boolean search(int x) {
        return false;
    }
}

File: AlgorithmGoVersion/src/DataStructure/heapStackQueue/queue/LinkedDequeuelj.go
package queue

type LinkDeQueuelj struct {
}

// Offer = push，入队
func (queue *LinkDeQueuelj) Offer(value int) bool {
	return false
}

// Poll 出队
func (queue *LinkDeQueuelj) Poll() int {
	return 0
}

// Peek 弹出第一个队首元素（不删除）
func (queue *LinkDeQueuelj) Peek() int {
	return 0
}

// Empty 队列判空
func (queue *LinkDeQueuelj) Empty() bool {
	return false
}

// GetRealSize 获取真实队列长度
func (queue *LinkDeQueuelj) GetRealSize() int {
	return 0
}

// GetMaxsize 获取最大队列长度
func (queue *LinkDeQueuelj) GetMaxsize() int {
	return 0
}

// ReSize 队列扩容
func (queue *LinkDeQueuelj) ReSize() {

}

// Search 查找函数
func (queue *LinkDeQueuelj) Search(value int) bool {
	return false
}

// offerFront offer = push, 队首入队
func (queue *LinkDeQueuelj) OfferFront(value int) bool {
	return false
}

// offerTail offer = push, 队尾入队
func (queue *LinkDeQueuelj) OfferTail(value int) bool {
	return false
}

// peekFront 弹出第一个队首元素（不删除）
func (queue *LinkDeQueuelj) PeekFront() int {
	return 0
}

// peekTail 弹出第一个队尾元素（不删除）
func (queue *LinkDeQueuelj) PeekTail() int {
	return 0
}

// pollFront poll = add, 队首出队
func (queue *LinkDeQueuelj) PollFront() int {
	return 0
}

// pollTail poll = add, 队尾出队
func (queue *LinkDeQueuelj) PollTail() int {
	return 0
}
